src/simulation/LoadGenerator.cpp

```diff
@@ -258,21 +262,69 @@ LoadGenerator::generateLoad(GeneratedLoadConfig cfg)
 
     for (int64_t i = 0; i < txPerStep; ++i)
     {
-        switch (cfg.mode)
+        if (cfg.mode == LoadGenMode::CREATE)
         {
-        case LoadGenMode::CREATE:
             cfg.nAccounts = submitCreationTx(cfg.nAccounts, cfg.offset,
                                              cfg.batchSize, ledgerNum);
+        }
+        else
+        {
+            auto sourceAccountId =
+                rand_uniform<uint64_t>(0, cfg.nAccounts - 1) + cfg.offset;
+
+            std::function<
+                std::pair<LoadGenerator::TestAccountPtr, TransactionFramePtr>()>
+                generateTx;
+
+            switch (cfg.mode)
+            {
+            case LoadGenMode::CREATE:
+                releaseAssert(false);
+                break;
+            case LoadGenMode::PAY:
+                generateTx = [&]() {
+                    return paymentTransaction(cfg.nAccounts, cfg.offset,
+                                              ledgerNum, sourceAccountId, 1,
+                                              cfg.maxGeneratedFeeRate);
+                };
+                break;
+            case LoadGenMode::PRETEND:
+            {
+                auto opCount = chooseOpCount(mApp.getConfig());
+                generateTx = [&, opCount]() {
+                    return pretendTransaction(cfg.nAccounts, cfg.offset,
+                                              ledgerNum, sourceAccountId,
+                                              opCount, cfg.maxGeneratedFeeRate);
+                };
+            }
             break;
-        case LoadGenMode::PAY:
-            cfg.nTxs = submitPaymentOrPretendTx(cfg, ledgerNum, 1);
-            break;
-        case LoadGenMode::PRETEND:
-            auto opCount = chooseOpCount(mApp.getConfig());
-            cfg.nTxs = submitPaymentOrPretendTx(cfg, ledgerNum, opCount);
+            case LoadGenMode::MIXED_TXS:
+            {
+                auto opCount = chooseOpCount(mApp.getConfig());
+                bool isDex = rand_uniform<uint32_t>(1, 100) <= cfg.dexTxPercent;
+                generateTx = [&, opCount, isDex]() {
+                    if (isDex)
+                    {
+                        return manageOfferTransaction(ledgerNum,
+                                                      sourceAccountId, opCount,
+                                                      cfg.maxGeneratedFeeRate);
+                    }
+                    else
+                    {
+                        return paymentTransaction(
+                            cfg.nAccounts, cfg.offset, ledgerNum,
+                            sourceAccountId, opCount, cfg.maxGeneratedFeeRate);
+                    }
+                };
+            }
             break;
-        }
+            }
 
+            if (submitTx(cfg, generateTx))
+            {
+                --cfg.nTxs;
```

Comment:
note that with this refactor, unlike the previous version, loadgen attempts to keep submitting transactions even after running into an error. I think this is fine @marta-lokhova can confirm

Reply:
Sorry I didn't see this earlier. I don't think we should keep submitting transactions after failures. We have some missions in CI that wait until loadgen succeeds or fails so the mission can "reset" core and retry. The expectation is that after completion no new tx load is generated by loadgen.

Reply:
Actually I didn't intend to change the behavior here; it was just a bug. Added a break on failure.